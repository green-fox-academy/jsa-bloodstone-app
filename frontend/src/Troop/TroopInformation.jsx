import React from 'react';
import {
  View, Image, Text,
  StyleSheet,
} from 'react-native';
import { CardView } from '../common/components';
import troopAvatar from '../../assets/troop/troop-avatar.jpg';
import attackIcon from '../../assets/troop/attack.png';
import defenceIcon from '../../assets/troop/defence.png';
import cookieIcon from '../../assets/troop/cookie.png';

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 15,
    lineHeight: 20,
  },
  troopAvatar: {
    width: 80,
    height: 80,
    marginRight: 24,
    borderRadius: 10,
  },
  fieldStyle: {
    flexDirection: 'row',
  },
  iconStyle: {
    width: 16,
    height: 16,
  },
});

function TroopInformation() {
  return (
    <CardView style={{ flexDirection: 'row', alignItems: 'center' }}>
      <Image style={styles.troopAvatar} source={troopAvatar} />
      <View>
        <View style={styles.fieldStyle}>
          <Text style={styles.textStyle}>Attack: 35</Text>
          <Image resizeMode="contain" source={attackIcon} style={styles.iconStyle} />
        </View>
        <View style={styles.fieldStyle}>
          <Text style={styles.textStyle}>Defence: 35</Text>
          <Image resizeMode="contain" source={defenceIcon} style={styles.iconStyle} />
        </View>
        <View style={styles.fieldStyle}>
          <Text style={styles.textStyle}>Sustenance: 23</Text>
          <Image resizeMode="contain" source={cookieIcon} style={styles.iconStyle} />
        </View>
      </View>
    </CardView>
  );
}

export default TroopInformation;
