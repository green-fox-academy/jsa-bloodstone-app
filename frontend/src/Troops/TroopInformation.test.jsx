import React from 'react';
import renderer from 'react-test-renderer';

import TroopInformation from './TroopInformation';

jest.mock('react-redux');
jest.mock('./actionCreator');

describe('<TroopInformation />', () => {
  it('Should render troop information and match the snapshot', () => {
    const component = renderer.create(<TroopInformation />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
