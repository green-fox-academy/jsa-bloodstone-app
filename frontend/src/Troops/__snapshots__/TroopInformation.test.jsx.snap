// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TroopInformation /> Should render troop information and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 10,
        "height": 80,
        "marginRight": 24,
        "width": 80,
      }
    }
  />
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "lineHeight": 20,
          }
        }
      >
        Attack: 10
      </Text>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "lineHeight": 20,
          }
        }
      >
        Defence: 10
      </Text>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "lineHeight": 20,
          }
        }
      >
        Sustenance: 10
      </Text>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;
